### ISSUE TITLE
The `globalThis` object is not defined in a collection context

### ISSUE THREAD

**Issue by countzero** [a85/POSTMan-Chrome-Extension#9319]
**Describe the bug**
The `globalThis` object is not defined in a collection context.

**To Reproduce**
1. Create a new collection
2. Create a GET request to "https://petstore.swagger.io/v2/pet/1"
3. Add the following line into the "Tests" field within the GET request: `console.log(globalThis);`
4. See error in console:

> ReferenceError: globalThis is not defined

**Expected behavior**
The [globalThis](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis) object is expected to be exposed in every JavaScript context within a collection.

**Screenshots**

![globalThis](https://user-images.githubusercontent.com/55295/100228110-77c93280-2f22-11eb-9a94-2a8295024828.png)

**App information:**
 - App Type: Native App
 - Postman Version: v7.36.0
 - OS: Windows 10

**Additional context**

This defect was not present in v7.34.0!

We are using the `globalThis` object to share frequently used functions within a collection across all requests. This seems to be the cleanest pattern and has already been used "in the wild": https://stackoverflow.com/a/62776695

```JavaScript
globalThis.myFunction = function(){}
```

There is nothing in the https://www.postman.com/downloads/release-notes/ that indicates that this change is a feature.

**Comment by DannyDainton** [comment]
I believe this is a duplicate of https://github.com/postmanlabs/postman-app-support/issues/9230 - Can you confirm @countzero

**Comment by countzero** [comment]
@DannyDainton: Yes that describes my problem, but I did not find that issue because I searched for the concrete word "globalThis".

**Comment by DannyDainton** [comment]
I'm going to close this one out and centralise all the information on there - We'll provide updates one we have them.

**Comment by codenirvana** [comment]
@countzero We never intended to expose the `globalThis` global in the postman script execution sandbox.
It was available for some time because of a bug which we fixed because of security reasons.

**Comment by countzero** [comment]
@codenirvana Thanks for getting back at me. What are you suggesting as a workaround for the following common use case: 

> **As a developer, I want to share frequently used functions across all requests within a collection, so that I do not have to copy paste that functions into each request and avoid human error.**

To give you an example on what we currently have to copy and paste across multiple requests:

``` JavaScript
// We can not place this functions globally until this bug is fixed: 
// https://github.com/postmanlabs/postman-app-support/issues/9319
const isString = value => typeof value === 'string';
const isInteger = value => Number.isInteger(value);
const isNull = value => value === null;
const isUndefined = value => typeof value === 'undefined';
const isUndefinedOrString = value => isUndefined(value) || isString(value);
const isUndefinedOrInteger = value => isUndefined(value) || isInteger(value);
const isNullOrString = value => isNull(value) || isString(value);
const isNullOrStringOrInteger = value => isNull(value) || isString(value) || isInteger(value);

// We can not place this function globally until this bug is fixed: 
// https://github.com/postmanlabs/postman-app-support/issues/9319
const expectMetadataType = (pm, value) => {
    pm.expect(value).to.be.an('object'); 
    pm.expect(value).to.have.own.property('metadata').that.is.an('object');
    pm.expect(value.metadata).to.have.own.property('client').that.is.a('string');
    pm.expect(value.metadata).to.have.own.property('clientAccount').that.is.a('string');
    pm.expect(value.metadata).to.have.own.property('userSystem').to.satisfy(isNullOrString);
    pm.expect(value.metadata).to.have.own.property('userId').to.satisfy(isNullOrStringOrInteger);
};
```
I would love to keep that stuff centralized in the collection context and only use it in the test itself:

``` JavaScript
pm.test('Response contains metadata type', () => {
    expectMetadataType(pm, pm.response.json());
});
```

I would like to see a replacement for `globalThis` in the form of i.e. a writable `pm.global` property. What do you think?

**Comment by codenirvana** [comment]
@countzero You can still set custom globals for this use case.

Assign global variable in collection pre-request script.
```js
util = {
    isString: value => typeof value === 'string',
    isInteger: value => Number.isInteger(value)
};

expectMetadataType = (pm, value) => {
    pm.expect(value).to.be.an('object'); 
};
```

And access `util.isString` or  `expectMetadataType` in other scripts.

**Comment by countzero** [comment]
@codenirvana  Thank you! That is exactly what I was looking for. I tried that, but prefixed my function with a `const`, which did not work! I took a second to reflect on why I failed:

1. That functionality is nowhere documented. Or at least I did not find it. I read completely through https://learning.postman.com/docs/writing-scripts and read every document that was listed for the search word "global"!

2. Omitting `var`, `let` or `const` is nothing a JavaScript programmer will try naturally because implicitly exposing a thing globally is a bad idea. That is why I used an explicit way via `globalThis` in the first place.

Are you planning on documenting this solution? 

What I also noted was that I can overwrite global objects like `postman` in the collection pre-request context, but that change will not be passed on to the request test context. Which is a good thing. But perhaps one should then use a very unique global object name that holds all custom functions to avoid naming conflicts in the future:

````JavaScript
const util = {
    isString: value => typeof value === 'string',
    isInteger: value => Number.isInteger(value),
};

const expectMetadataType = (pm, value) => {
    pm.expect(value).to.be.an('object'); 
};

// Expose global functions under the unique company namespace "stadtwerk".
stadtwerk = {
    util,
    expectMetadataType,
};

````
Which then can be used in the test scope like:

```` JavaScript
stadtwerk.util.isString('Hello World!');
````
